src/components/AssetFormHeader.js

```diff
@@ -6,35 +6,78 @@ import AppBar from 'material-ui/AppBar';
 import Toolbar from 'material-ui/Toolbar';
 import Typography from 'material-ui/Typography';
 import Button from 'material-ui/Button';
+import {connect} from "react-redux";
 
 const styles = theme => ({
   flex: { flex: 1 },
   cancelButton: { color: theme.customColors.white },
   saveButton: { background: theme.customColors.white }
 });
 
-/*
-  Renders the app bar of the form for the creation/editing of an Asset.
-  */
-const AssetFormHeader = ({ title, onClick, classes }) => (
-  <AppBar position="static">
-    <Toolbar>
-      <Typography variant="title" color="inherit" className={classes.flex}>
-        { title }
-      </Typography>
+/**
+ * Renders the app bar of the form for the creation/editing/viewing of an Asset.
+ *
+ * @param onClick handles the save button
+ * @param asset either the asset being edited/viewed or null if we are creating an asset
+ * @param ownsAsset closure that decides if the user owns an asset.
+ */
+const AssetFormHeader = ({ onClick, asset, ownsAsset, classes }) => {
+
+  // the default title/buttons are for creating an asset
+  let title = 'Create new asset';
+  let buttons = (
+    <div>
       <Link className='App-raised-button-link' to="/">
         <Button className={classes.cancelButton} color='primary'>Cancel</Button>
       </Link>
       &nbsp;
       <Button className={classes.saveButton} variant="raised" onClick={onClick}>Save</Button>
+    </div>
+  );
+
+  if (asset) {
+    // define the asset name
+    const name = asset.name ? asset.name : asset.id;
+
+    if (ownsAsset(asset)) {
+      title = 'Editing: ' + name;
+    } else {
+      title = 'Viewing: ' + name;
+      // overrides the default buttons with a simple back button
+      buttons = (
+        <Link className='App-raised-button-link' to="/">
+          <Button className={classes.cancelButton} color='primary'>Back</Button>
+        </Link>
+      );
+    }
+  }
+
+  return <AppBar position="static">
+    <Toolbar>
+      <Typography variant="title" color="inherit" className={classes.flex}>
+        { title }
+      </Typography>
+      {buttons}
     </Toolbar>
-  </AppBar>
-);
+  </AppBar>;
+};
 
 AssetFormHeader.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  asset: PropTypes.object,
   classes: PropTypes.object.isRequired,
-  title: PropTypes.string.isRequired,
-  onClick: PropTypes.func,
 };
 
-export default withStyles(styles)(AssetFormHeader);
+const mapStateToProps = ({ lookupApi: { self } }) => {
+
+  const institutions = self && self.institutions ? self.institutions : [];
+
+  // closure that decides if the user owns an asset.
+  const ownsAsset = (asset) => (
```

Comment:
So, AssetFormHeader gets asset as a prop. Why can't it just set a boolean directly? E.g.
```js
const mapStateToProps = ({ lookupApi: { self } }) => {
   // ...
   return { ownsAsset: institutions.find((institution) => institution.instid === asset.department) }
}
```